dmn-engine/src/main/scala/org/camunda/dmn/evaluation/InvocationEvaluator.scala

```diff
@@ -24,15 +24,18 @@ class InvocationEvaluator(
   def eval(invocation: ParsedInvocation,
            context: EvalContext): Either[Failure, Val] = {
 
-    evalParameters(invocation.bindings, context).right.flatMap { p =>
+    evalParameters(invocation.bindings, context).flatMap { p =>
       val ctx = context.copy(variables = context.variables ++ p.toMap)
 
-      evalBkm(invocation.invocation, ctx).right
-        .map { result =>
+      evalBkm(invocation.invocation, ctx) match {
+        case r @ Right(result) =>
           context.audit(invocation, SingleEvaluationResult(result))
-
-          result
-        }
+          r
+        case l @ Left(failure) =>
+          context.audit(invocation,
+                        SingleEvaluationResult(ValError(failure.message)))
+          l
+      }
```

Comment:
With the suggested audit function, this can be simplified to:

```
    evalParameters(invocation.bindings, context).flatMap { p =>
      val ctx = context.copy(variables = context.variables ++ p.toMap)

      val result = evalBkm(invocation.invocation, ctx)
      context.audit(invocation, result)
      result
    }
```